hera_sim/simulate.py

```diff
@@ -434,4 +435,64 @@ def add_xtalk(self, model='gen_whitenoise_xtalk', bls=None, **kwargs):
                 vis=self.data.data_array[blt_ind, 0, :, pol_ind],
                 xtalk=xtalk
             )
+    
+    def make_data(self, sim_file=None, sim_params=None):
+        """
+        Accept a dictionary or YAML file of simulation parameters and add in
+        all of the desired simulation components to the Simulator object.
+
+        Args:
+            sim_file (str): string providing a hook to a YAML file
+            
+            sim_params (dict): dictionary of simulation parameters.
+                it must take the form {model:params}, where model is
+                a string specifying the simulation component to be added
+                and params is a dictionary providing all the keyword
+                arguments and their desired values.
+        """
+        # make a dictionary whose values point to the various methods
+        # used to add different simulation components
+        SIMULATION_COMPONENTS = { 'noiselike_eor':self.add_eor,
+                                  'diffuse_foreground':self.add_foreground,
+                                  'pntsrc_foreground':self.add_foreground,
+                                  'gains':self.add_gains,
+                                  'sigchain_reflections':self.add_sigchain_reflections,
+                                  'gen_whitenoise_xtalk':self.add_xtalk,
+                                  'gen_cross_coupling_xtalk':self.add_xtalk,
+                                  'thermal_noise':self.add_noise }
+
+        # make a tuple keeping track of which components do not use a model
+        uses_no_model = ('gains', 'sigchain_reflections')
+
+        assert sim_file is not None or sim_params is not None, \
+                'Either a hook to a simulation file or a dictionary of ' + \
+                'simulation parameters must be provided.'
+
+        # if a hook to a simulation file is provided, then read it in
+        # XXX is there a way to check that the formatting of sim_file is OK?
```

Comment:
I admit that I'm not exactly sure what you mean by this, since I was referencing `sim_file`. That said, I'm not exactly sure what i was getting at with the question of checking the formatting of the YAML file. I was mainly concerned about the safety of `yaml`'s `FullLoader`, although if I remember correctly, the developers state that the `FullLoader` prevents execution of code.

Reply:
Is it that you were suggesting I `try` to load in the contents of `sim_file` and do something else if the `except` clause is executed?

Reply:
yeah, that's what I thought you meant. Yes, code shouldn't be executed when using `FullLoader`, so that's all good. Admittedly, using this `try: except:` doesn't really enforce proper formatting of the YAML file -- it means its valid _YAML_, but not necessarily that it contains the right kinds of keywords for this particular case. 

I think your solution is good.